elastic/beats
# [Metricbeat] Add STAN (NATS streaming) module

This PR cherry-picks the commits of @devon-kim 
 from https://github.com/elastic/beats/compare/7.2...devon-kim:ops-1844-stan-metricbeat so as to be merged to `beats/master` as part of https://github.com/elastic/beats/issues/14629.


cc: @exekias

### Manual testing
1. Build the Docker image: `docker build -t stan_test .` (run this command under `x-pack/metricbeat/module/stan/_meta`)
2. Start STAN server: `docker run -p 8222:8222 stan_test`
3. Enable the module and run Metricbeat. 
(Note that default fetching period is 60secs, so this might should be changed for the testing)
